netmon.cc

```diff
@@ -362,35 +361,42 @@ static unsigned short internetchecksum(void *b, int len)
 	return result;
 }
 
-static std::string makeICMPQuery(int family, uint16_t id, uint16_t seq)
+static std::string makeICMPQuery(int family, uint16_t id, uint16_t seq, size_t psize)
 {
   if(family==AF_INET) {
-    icmppacket p;
-    memset(&p, 0, sizeof(p));
-    p.hdr.type = ICMP_ECHO;
-    p.hdr.un.echo.id = id;
-    p.hdr.un.echo.sequence = seq;
+    size_t full_size = sizeof(icmphdr) + psize;
+    icmppacket *p = (icmppacket*)malloc(full_size);
```

Comment:
I'm not too hot on naked calls to malloc, and it appears we also lack a call to free? could you perhaps try:

```C++
vector<char> store(full_size);
icmppacket *p = (icmppacket*) &store[0];
```
And see if that works?

Reply:
Indeed, sorry was late hours, and forgot to free. So better to avoid dynamic allocation at all, did as you suggested and amended commit.